wls-gui-wahllokalsystem/src/composables/user/userMapper.ts

```diff
@@ -1,37 +1,114 @@
 import type { User } from "@/types/User.ts";
+import type { WahlMetaData } from "@/types/wlsTypes/WahlMetaData.ts";
 
 export function useUserMapper() {
-  function toModel(userDto: UserDTO): User {
-    const user: User = {
-      username: userDto.username ?? "",
-      email: userDto.email ?? "",
-      userEnabled: userDto.userEnabled ?? false,
-      wahltagID: userDto.wahltagID ?? "",
-      wahltag: userDto.wahltag ?? "",
-      wahlbezirkID: userDto.wahlbezirkID ?? "",
-      wahlbezirkNummer: userDto.wahlbezirkNummer ?? "",
-      wahlbezirksArt: userDto.wahlbezirksArt ?? undefined,
-      pin: userDto.pin ?? "",
-      authorities: userDto.authorities ?? new Set<string>(),
-      wahlMetaData: undefined,
-    };
-
-    try {
-      const parsed_wbid_wahlnummer = userDto.wbid_wahlnummer
-        ? JSON.parse(userDto.wbid_wahlnummer)
-        : undefined;
-
-      user.wahlMetaData = parsed_wbid_wahlnummer?.wbid_wahlnummer
-        ? parsed_wbid_wahlnummer.wbid_wahlnummer
-        : undefined;
-    } catch {
-      console.debug("failed to parse JSON wbid_wahlnummer");
-      user.wahlMetaData = undefined;
+  function validateDtoAndMapToModel(userDto: UserDTO): User {
+    const errors: Error[] = [];
+
+    _assertIsNotBlank(userDto.username, "username", errors);
+    _assertIsNotUndefinedOrNull(userDto.userEnabled, "userEnabled", errors);
+    _assertIsNotBlank(userDto.wahltagID, "wahltagID", errors);
+    _assertIsNotBlank(userDto.wahltag, "wahltag", errors);
+    _assertIsNotBlank(userDto.wahlbezirkID, "wahlbezirkID", errors);
+    _assertIsNotBlank(userDto.wahlbezirkNummer, "wahlbezirkNummer", errors);
+    _assertIsNotBlank(userDto.wahlbezirksArt, "wahlbezirksArt", errors);
+    _assertIsNotBlank(userDto.pin, "pin", errors);
+    _assertIsNotEmpty(userDto.authorities, "authorities", errors);
+    _assertIsNotBlank(userDto.wbid_wahlnummer, "wbid_wahlnummer", errors);
+
+    if (errors.length > 0) {
+      throw new Error(
+        `Laden des Users fehlgeschlagen. Folgende Pflichtfelder sind nicht befüllt: ${errors.join(", ")}`
+      );
+    }
+
+    return {
+      username: userDto.username,
+      email: userDto.email,
+      userEnabled: userDto.userEnabled,
+      wahltagID: userDto.wahltagID,
+      wahltag: userDto.wahltag,
+      wahlbezirkID: userDto.wahlbezirkID,
+      wahlbezirkNummer: userDto.wahlbezirkNummer,
+      wahlbezirksArt: userDto.wahlbezirksArt,
+      pin: userDto.pin,
+      authorities: userDto.authorities,
+      wahlMetaData: _parseWbIdWahlnummer(userDto.wbid_wahlnummer),
+    } as User;
+  }
+
+  function _assertIsNotBlank(
+    value: string | undefined | null,
+    propName: string,
+    errors?: Error[]
+  ): asserts value is string {
+    if (value === undefined || value == null || value.trim() === "") {
+      if (errors) {
+        errors.push(new Error(`${propName} is blank`));
+      } else {
+        throw new Error(`${propName} is blank`);
+      }
     }
+  }
```

Comment:
Das mit dem optionalen war nur eine fixe Idee in meinem Entwurf. Würde ich zur Vereinfachung entfernen. Bitte auf für die anderen checks übernehmen.

```suggestion
  function _assertIsNotBlank(
    value: string | undefined | null,
    propName: string,
    errors: Error[]
  ): asserts value is string {
    if (value === undefined || value == null || value.trim() === "") {
      errors.push(new Error(`${propName} is blank`));
    }
  }
```

Reply:
[dd02291](https://github.com/it-at-m/Wahllokalsystem/pull/1450/commits/dd0229164c196e1e75b19bdae057f24070d85e80)